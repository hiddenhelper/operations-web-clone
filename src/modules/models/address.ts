import { INamedEntity } from './general';

export enum AddressFields {
  LINE_1 = 'line1',
  LINE_2 = 'line2',
  COUNTRY = 'country',
  COUNTRY_ID = 'countryId',
  CITY = 'city',
  STATE = 'stateCode',
  STATE_NAME = 'stateName',
  ZIP_CODE = 'zipCode',
  COUNTY = 'county',
  BOROUGH = 'borough',
  LATITUDE = 'latitude',
  LONGITUDE = 'longitude',
  ATTENTION_1 = 'attention1',
  ATTENTION_2 = 'attention2',
}

export enum LocationsFields {
  NAME = 'name',
}

export enum Borough {
  MANHATTAN = 'Manhattan',
  BROOKLYN = 'Brooklyn',
  QUEENS = 'Queens',
  STATEN_ISLAND = 'Staten Island',
  THE_BRONX = 'The Bronx',
}

export interface IAddress {
  [AddressFields.LINE_1]: string;
  [AddressFields.LINE_2]: string;
  [AddressFields.CITY]: string;
  [AddressFields.STATE]: string;
  [AddressFields.ZIP_CODE]: string;
  [AddressFields.COUNTRY]?: INamedEntity;
  [AddressFields.COUNTRY_ID]?: string;
  [AddressFields.COUNTY]?: string;
  [AddressFields.ATTENTION_1]?: string;
  [AddressFields.ATTENTION_2]?: string;
  [AddressFields.STATE_NAME]?: string;
  [AddressFields.BOROUGH]?: Borough;
  [AddressFields.LATITUDE]: number;
  [AddressFields.LONGITUDE]: number;
}

export interface ILocation {
  id?: string;
  [LocationsFields.NAME]?: string;
  isUsed?: boolean;
}

export const addressFieldRules = {
  required: [
    { name: AddressFields.LINE_1, required: true },
    { name: AddressFields.LINE_2, required: false },
    { name: AddressFields.COUNTRY_ID, required: true, computePositive: true },
    { name: AddressFields.STATE, required: true },
    { name: AddressFields.ZIP_CODE, required: true },
    { name: AddressFields.CITY, required: true },
    { name: AddressFields.ATTENTION_1, required: false },
    { name: AddressFields.ATTENTION_2, required: false },
  ],
  notRequired: [
    { name: AddressFields.LINE_1, required: false },
    { name: AddressFields.LINE_2, required: false },
    { name: AddressFields.COUNTRY_ID, required: false },
    { name: AddressFields.STATE, required: false },
    { name: AddressFields.ZIP_CODE, required: false },
    { name: AddressFields.CITY, required: false },
    { name: AddressFields.ATTENTION_1, required: false },
    { name: AddressFields.ATTENTION_2, required: false },
  ],
};

export const boroughOptionList = [
  {
    value: Borough.MANHATTAN,
    label: Borough.MANHATTAN,
  },
  {
    value: Borough.BROOKLYN,
    label: Borough.BROOKLYN,
  },
  {
    value: Borough.QUEENS,
    label: Borough.QUEENS,
  },
  {
    value: Borough.STATEN_ISLAND,
    label: Borough.STATEN_ISLAND,
  },
  {
    value: Borough.THE_BRONX,
    label: Borough.THE_BRONX,
  },
];

export enum ZipCodeFormatType {
  FIVE_DIGITS = 0,
  FIVE_DIGITS_PLUS_FOUR = 1,
  CANADA_FORMAT = 2,
}

export const zipCodeFormatMap: { [key: string]: { placeholder: string; mask: any[] } } = {
  [ZipCodeFormatType.FIVE_DIGITS]: {
    placeholder: '00000',
    mask: [/\d/, /\d/, /\d/, /\d/, /\d/, /\d/],
  },
  [ZipCodeFormatType.FIVE_DIGITS_PLUS_FOUR]: {
    placeholder: '00000-0000',
    mask: [/\d/, /\d/, /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  },
  [ZipCodeFormatType.CANADA_FORMAT]: {
    placeholder: 'ANA-NAN',
    mask: [/[a-zA-Z]/, /\d/, /[a-zA-Z]/, '-', /\d/, /[a-zA-Z]/, /\d/],
  },
};

export const getFallbackAddress = () => ({
  line1: null,
  line2: null,
  city: null,
  countryId: null,
  stateCode: null,
  zipCode: null,
  county: null,
  latitude: null,
  longitude: null,
  stateName: null,
});

export const getFallbackLocation = (): ILocation => ({
  name: null,
});
