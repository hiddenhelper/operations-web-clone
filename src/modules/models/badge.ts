import { INamedEntity } from './general';
import { WorkerProjectStatus } from './project';

export enum BadgeStatus {
  ACTIVE = 0,
  REVOKED = 1,
  DEACTIVATE = 2,
  EXPIRED = 3,
  NOT_ISSUED = 4,
}

export enum BadgeLayout {
  LARGE_HEADER = 1,
  SMALL_HEADER = 2,
  WHITE_HEADER = 3,
  VISITOR = 4,
}

export enum BadgeType {
  REGULAR = 0,
  UNIVERSAL = 1,
}

export enum VisitorAvailability {
  PENDING = 0,
  AVAILABLE = 1,
  ASSIGNED = 2,
}

export enum VisitorType {
  REGULAR = 0,
  VIP = 1,
}

export enum VisitorEntity {
  CLIENT = 0,
  OTHER = 1,
}

export enum VisitorBadgeStatus {
  AVAILABLE = 0,
  ASSIGNED = 1,
}

export interface IBadgeTemplate {
  id: string;
  layout: BadgeLayout;
  color: string;
  fontColor: string;
  font: string;
  logoUrl?: string;
  templateFileUrl?: string;
}

export interface IBadge {
  id: string;
  badgeTemplate: IBadgeTemplate;
  code: string;
  status: BadgeStatus;
  firstName: string;
  lastName: string;
  emergencyContactName: string;
  emergencyContactPhone: string;
  emergencyContactRelationship: string;
  pictureUrl: string;
  companyName: string;
  hardHatNumber: string;
  tagId?: string;
  expirationDate?: string;
  badgeType?: BadgeType;
  consentFormSigned?: boolean;
  trainingsCompleted?: boolean;
  certificationsCompleted?: boolean;
  defaultExpirationDate?: string;
  workerProjectStatus?: WorkerProjectStatus;
  isSynchronizing?: boolean;
}

export enum BadgeAction {
  UNKNOWN = 0,
  ACTIVATE = 1,
  DEACTIVATE = 2,
  REVOKE = 3,
  REACTIVATE = 4,
  ADD_TO_ACS = 5,
  REMOVE_FROM_ACS = 6,
  UPDATE_EXPIRATION_DATE = 7,
  GENERATE_PDF = 8,
  INSERT_TAG_ID = 9,
}

export interface IBadgeHistory {
  id: string;
  date: string;
  badgeAction: BadgeAction;
  tagId: string;
  description: string;
  user: {
    id: string;
    firstName: string;
    lastName: string;
    email: string;
  };
  projectAccessControlSystem: {
    id: string;
    deviceName: string;
  };
  reader: {
    id: string;
    hostname: string;
  };
}

export interface IBadgeUpdateRequest {
  expirationDate?: string;
  hasExpiration?: number;
  tagId?: string;
}

export interface IBadgeVisitor {
  id: string;
  number: number;
  code: string;
  status: BadgeStatus;
  availability: VisitorAvailability;
  firstName: string;
  lastName: string;
  visitorType: VisitorType;
  validFrom: string;
  validTo: string;
  description: string;
  company?: INamedEntity;
  companyId?: string;
  entityName: string;
  visitorBadgeStatus: VisitorBadgeStatus;
  tagId?: string;
  isSynchronizing?: boolean;
}

export const badgeActionMap = {
  [BadgeAction.UNKNOWN]: () => 'Unknown',
  [BadgeAction.ACTIVATE]: () => 'Activate the badge',
  [BadgeAction.DEACTIVATE]: () => 'Deactivate the badge',
  [BadgeAction.REVOKE]: () => 'Revoke the badge',
  [BadgeAction.REACTIVATE]: () => 'Reactivate the badge',
  [BadgeAction.ADD_TO_ACS]: (name: string) => `Badge added on ACS ${name}`,
  [BadgeAction.REMOVE_FROM_ACS]: (name: string) => `Badge removed on ACS ${name}`,
  [BadgeAction.UPDATE_EXPIRATION_DATE]: () => 'Change the expiration date',
  [BadgeAction.GENERATE_PDF]: () => 'Generate PDF',
  [BadgeAction.INSERT_TAG_ID]: () => 'Insert Tag ID',
};

export const badgeOptionList = [{ title: 'Deactivate' }, { title: 'Revoke' }, { title: 'Reprint' }];

export const badgeStatusMap = {
  [BadgeStatus.ACTIVE]: 'Active',
  [BadgeStatus.REVOKED]: 'Revoked',
  [BadgeStatus.DEACTIVATE]: 'Deactivated',
  [BadgeStatus.EXPIRED]: 'Expired',
  [BadgeStatus.NOT_ISSUED]: 'Not Issued',
};

export const modalStatusMap = {
  [BadgeStatus.ACTIVE]: () => ({
    title: 'Activate Badge?',
    text: `Enter Badge Tag ID`,
    confirmLabel: 'Activate',
    loadingText: 'Activating...',
  }),
  [BadgeStatus.DEACTIVATE]: () => ({
    title: 'Deactivate Badge?',
    text: 'Please enter the reason to deactivate the Badge.',
    confirmLabel: 'Deactivate',
    loadingText: 'Deactivating...',
  }),
  [BadgeStatus.REVOKED]: () => ({
    title: 'Revoke Badge?',
    text: 'Please enter the reason to revoke the Badge.',
    confirmLabel: 'Revoke',
    loadingText: 'Revoking...',
  }),
  [BadgeStatus.EXPIRED]: () => ({}),
};

export const visitorAvailabilityMap = {
  [VisitorAvailability.AVAILABLE]: 'Available',
  [VisitorAvailability.ASSIGNED]: 'Assigned',
};

export const visitorTypeMap = {
  [VisitorType.REGULAR]: 'Regular',
  [VisitorType.VIP]: 'VIP',
};

export const visitorEntityMap = {
  [VisitorEntity.CLIENT]: 'Client',
  [VisitorEntity.OTHER]: 'Other',
};

export const entityOptionList = [
  { value: VisitorEntity.CLIENT, label: visitorEntityMap[VisitorEntity.CLIENT] },
  { value: VisitorEntity.OTHER, label: visitorEntityMap[VisitorEntity.OTHER] },
];

export enum BadgeModalAction {
  ACTIVE = 0,
  NOT_ACTIVE = 1,
}

export const getFallbackBadgeTemplate = (): IBadgeTemplate => ({
  id: null,
  layout: BadgeLayout.LARGE_HEADER,
  color: '1E86E8',
  fontColor: '444444',
  font: 'Arial',
  logoUrl: null,
  templateFileUrl: null,
});

export const getFallbackVisitorBadgeTemplate = (): IBadgeTemplate => ({
  id: null,
  layout: BadgeLayout.VISITOR,
  color: 'FFE304',
  fontColor: '444444',
  font: 'Arial',
  logoUrl: null,
  templateFileUrl: null,
});

export const getFallbackBadge = (isVisitor = false): IBadge => ({
  id: null,
  badgeTemplate: isVisitor ? getFallbackVisitorBadgeTemplate() : getFallbackBadgeTemplate(),
  status: BadgeStatus.DEACTIVATE,
  firstName: null,
  lastName: null,
  emergencyContactName: null,
  emergencyContactPhone: null,
  emergencyContactRelationship: null,
  companyName: null,
  hardHatNumber: null,
  code: null,
  pictureUrl: null,
  tagId: null,
  expirationDate: null,
  consentFormSigned: false,
  trainingsCompleted: false,
  certificationsCompleted: false,
  badgeType: BadgeType.REGULAR,
  defaultExpirationDate: null,
  workerProjectStatus: WorkerProjectStatus.PENDING,
});

export const getFallbackBadgeVisitor = (): IBadgeVisitor => ({
  id: null,
  number: null,
  code: null,
  firstName: null,
  lastName: null,
  validFrom: null,
  validTo: null,
  companyId: null,
  entityName: null,
  description: null,
  visitorType: VisitorType.REGULAR,
  status: BadgeStatus.DEACTIVATE,
  availability: VisitorAvailability.AVAILABLE,
  visitorBadgeStatus: VisitorBadgeStatus.AVAILABLE,
  tagId: null,
});
