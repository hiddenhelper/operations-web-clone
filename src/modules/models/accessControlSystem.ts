import { ILocation } from './address';
import { IDevice, DeviceStatus } from './device';

export enum AccessControlSystemType {
  PORTAL = 0,
  TURNSTILE = 1,
  ENCLOSED_TS = 2,
  HANDHELD = 3,
}

export enum ReaderType {
  Entrance = 0,
  Exit = 1,
  BiDirectionalByAntenna = 2,
  BiDirectionalBySpec = 3,
}

export interface IReader {
  serialNumber: string;
  lastMaintenanceDate?: string;
  inServiceDate: string;
  model: string;
  notes?: string;
  hostname?: string;
  directionType?: ReaderType;
  hostAddress?: string;
  telnetConnectionPort?: string;
  sshConnectionPort?: string;
  httpConnectionPort?: string;
  tcpConnectionPort?: string;
}

export enum LifeCycleType {
  NEW = 0,
  UPGRADE = 1,
  IN_FIELD_TRANSFER = 2,
  REFURBISHED = 3,
}

export enum AccessControlSystemVersion {
  V1 = 0,
  V2 = 1,
  V3 = 2,
  V3_FACIAL = 3,
}

export interface IAccessControlSystem extends IDevice {
  type: AccessControlSystemType;
  lifeCycle: LifeCycleType;
  lastRefurbishedDate: string;
  lastMaintenanceDate: string;
  inServiceDate: string;
  warrantyExpirationDate: string;
  orderDate?: string;
  vendor?: string;
  invoice?: string;
  reader1?: IReader;
  reader2?: IReader;
  version: AccessControlSystemVersion;
  project?: {
    id: string;
    name: string;
  };
  status: DeviceStatus;
  location?: ILocation;
  hasReverseInstallation?: boolean;
  reader1SerialNumber?: string;
  reader2SerialNumber?: string;
  reader1Hostname?: string;
  reader2Hostname?: string;
}

export interface IProjectAccessControlSystem {
  id?: string;
  accessControlSystemId?: string;
  projectId?: string;
  type: AccessControlSystemType;
  serialNumber: string;
  locationId?: string;
  location?: ILocation;
  deviceName: string;
  hasReverseInstallation: boolean;
  reader1: IReader;
  reader2?: IReader;
}

export interface IProjectAccessControlSystemByLocation {
  location?: ILocation;
  accessControlSystems: IAccessControlSystem[];
}

export const accessControlSystemTypeMap = {
  [AccessControlSystemType.ENCLOSED_TS]: 'Enclosed TS',
  [AccessControlSystemType.HANDHELD]: 'Handheld',
  [AccessControlSystemType.PORTAL]: 'Open Portal',
  [AccessControlSystemType.TURNSTILE]: 'Turnstile',
};

export const typeOptionList = [
  { value: AccessControlSystemType.PORTAL, label: accessControlSystemTypeMap[AccessControlSystemType.PORTAL] },
  { value: AccessControlSystemType.TURNSTILE, label: accessControlSystemTypeMap[AccessControlSystemType.TURNSTILE] },
  { value: AccessControlSystemType.ENCLOSED_TS, label: accessControlSystemTypeMap[AccessControlSystemType.ENCLOSED_TS] },
  { value: AccessControlSystemType.HANDHELD, label: accessControlSystemTypeMap[AccessControlSystemType.HANDHELD] },
];

export const lifeCycleMap = {
  [LifeCycleType.NEW]: 'New',
  [LifeCycleType.UPGRADE]: 'Upgrade',
  [LifeCycleType.IN_FIELD_TRANSFER]: 'In Field Transfer',
  [LifeCycleType.REFURBISHED]: 'Refurbished',
};

export const lifeCycleOptionList = [
  { value: LifeCycleType.NEW, label: lifeCycleMap[LifeCycleType.NEW] },
  { value: LifeCycleType.UPGRADE, label: lifeCycleMap[LifeCycleType.UPGRADE] },
  { value: LifeCycleType.IN_FIELD_TRANSFER, label: lifeCycleMap[LifeCycleType.IN_FIELD_TRANSFER] },
  { value: LifeCycleType.REFURBISHED, label: lifeCycleMap[LifeCycleType.REFURBISHED] },
];

export const deviceVersionMap = {
  [AccessControlSystemVersion.V1]: 'V1',
  [AccessControlSystemVersion.V2]: 'V2',
  [AccessControlSystemVersion.V3]: 'V3',
  [AccessControlSystemVersion.V3_FACIAL]: 'V3Facial',
};

export const deviceVersionOptionList = [
  { value: AccessControlSystemVersion.V1, label: deviceVersionMap[AccessControlSystemVersion.V1] },
  { value: AccessControlSystemVersion.V2, label: deviceVersionMap[AccessControlSystemVersion.V2] },
  { value: AccessControlSystemVersion.V3, label: deviceVersionMap[AccessControlSystemVersion.V3] },
  { value: AccessControlSystemVersion.V3_FACIAL, label: deviceVersionMap[AccessControlSystemVersion.V3_FACIAL] },
];

export const readerTypeMap = {
  [ReaderType.Entrance]: 'Entrance',
  [ReaderType.Exit]: 'Exit',
  [ReaderType.BiDirectionalByAntenna]: 'BiDirectional By Antenna',
  [ReaderType.BiDirectionalBySpec]: 'BiDirectional By Spec',
};

export const readerTypeByTypeMap = {
  [AccessControlSystemType.PORTAL]: [
    { label: readerTypeMap[ReaderType.Entrance], value: ReaderType.Entrance },
    { label: readerTypeMap[ReaderType.Exit], value: ReaderType.Exit },
    { label: readerTypeMap[ReaderType.BiDirectionalByAntenna], value: ReaderType.BiDirectionalByAntenna },
  ],
  [AccessControlSystemType.TURNSTILE]: [{ label: readerTypeMap[ReaderType.BiDirectionalBySpec], value: ReaderType.BiDirectionalBySpec }],
  [AccessControlSystemType.ENCLOSED_TS]: [{ label: readerTypeMap[ReaderType.BiDirectionalBySpec], value: ReaderType.BiDirectionalBySpec }],
  [AccessControlSystemType.HANDHELD]: [],
};

export const getFallbackAccessControlSystem = (): IAccessControlSystem => ({
  id: null,
  type: AccessControlSystemType.PORTAL,
  serialNumber: null,
  lifeCycle: LifeCycleType.NEW,
  lastRefurbishedDate: null,
  lastMaintenanceDate: null,
  inServiceDate: null,
  warrantyExpirationDate: null,
  orderDate: null,
  vendor: null,
  invoice: null,
  project: null,
  status: null,
  notes: null,
  version: AccessControlSystemVersion.V1,
  reader1: getFallbackReader(),
  reader2: null,
  hasReverseInstallation: false,
});

export const getFallbackReader = (): IReader => ({
  serialNumber: null,
  lastMaintenanceDate: null,
  inServiceDate: null,
  model: null,
  notes: null,
});

export const getFallbackProjectAccessControlSystem = (): IProjectAccessControlSystem => ({
  type: AccessControlSystemType.PORTAL,
  accessControlSystemId: null,
  serialNumber: null,
  deviceName: null,
  hasReverseInstallation: false,
  reader1: null,
  reader2: null,
});
