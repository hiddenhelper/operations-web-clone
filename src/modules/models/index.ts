import * as GeneralModel from './general';
import * as UserModel from './user';
import * as ClientModel from './client';
import * as ProjectModel from './project';
import * as ResourceModel from './resource';
import * as WorkerModel from './worker';
import * as AccessControlSystemModel from './accessControlSystem';
import * as BadgePrintingSystemModel from './badgePrinterSystem';
import * as DeviceModel from './device';
import * as InventoryModel from './inventory';
import * as BadgeModel from './badge';
import * as CertificationModel from './certification';
import * as FileModel from './file';
import * as AddressModel from './address';
import * as ConsentFormModel from './consentForm';
import * as StatisticsModel from './statistics';
import * as TrainingModel from './training';
import * as PaymentModel from './payment';
import * as InvoiceModel from './invoice';
import * as ProcoreModel from './procore';
import * as SearchModel from './search';
import * as ProjectNewModel from './project-new';

export {
  GeneralModel,
  UserModel,
  ClientModel,
  ProjectModel,
  ResourceModel,
  AccessControlSystemModel,
  WorkerModel,
  BadgePrintingSystemModel,
  DeviceModel,
  InventoryModel,
  BadgeModel,
  CertificationModel,
  FileModel,
  AddressModel,
  ConsentFormModel,
  StatisticsModel,
  TrainingModel,
  PaymentModel,
  InvoiceModel,
  ProcoreModel,
  SearchModel,
  ProjectNewModel,
};
