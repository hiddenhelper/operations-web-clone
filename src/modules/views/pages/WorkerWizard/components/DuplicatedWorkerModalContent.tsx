import React, { memo, useCallback } from 'react';
import moment from 'moment';

import Typography from '@material-ui/core/Typography';
import PersonIcon from '@material-ui/icons/Person';
import Avatar from '@material-ui/core/Avatar';

import MatchingField, { MatchType } from '../../../shared/MatchingField/MatchingField';

import { UserModel, WorkerModel } from '../../../../models';

import { formatPhoneNumber, getConditionalDefaultValue } from '../../../../../utils/generalUtils';
import { avatarGlobalStyles, modalGlobalStyles } from '../../../../../assets/styles';
import { useStyles } from '../styles';

export interface IExistingWorkerModalProps {
  matchedFields: string[];
  existingWorker: WorkerModel.IExistingWorker;
  currentWorker: WorkerModel.IWorker;
  userRole: UserModel.Role;
}

const DuplicatedWorkerModalContent = ({ existingWorker, matchedFields, currentWorker, userRole }: IExistingWorkerModalProps) => {
  const modalClasses = modalGlobalStyles();
  const avatarGlobalClasses = avatarGlobalStyles();
  const classes = useStyles();

  const getMatchType = useCallback(
    (fieldName: string) => {
      return getConditionalDefaultValue(matchedFields.includes(fieldName), MatchType.MATCH, MatchType.NO_MATCH);
    },
    [matchedFields]
  );

  return (
    <div className={classes.duplicatedWorkerModalContent}>
      <Typography className={`${classes.workerModalSecondaryTextPosition} ${modalClasses.secondaryText}`}>
        This information matches with an existing worker.
      </Typography>
      <div className={classes.avatarWrapper}>
        <div>
          <Avatar
            className={`${classes.workerModalAvatarPosition} ${avatarGlobalClasses.avatarWrapper} ${avatarGlobalClasses.workerMedium}`}
            src={existingWorker.pictureUrl}
            alt="Worker profile"
          >
            <PersonIcon />
          </Avatar>
        </div>
        <span>
          <ul className={classes.listWrapper}>
            <MatchingField type={getMatchType('fullName')}>
              <Typography className={classes.workerAccent}>{existingWorker.fullName}</Typography>
            </MatchingField>
            {existingWorker.companyName && userRole === UserModel.Role.FCA_ADMIN && (
              <MatchingField type={getMatchType('companyName')}>
                <Typography className={classes.companyAccent}>{existingWorker.companyName}</Typography>
              </MatchingField>
            )}
            <MatchingField type={getMatchType('dateOfBirth')}>Date of Birth: {moment(existingWorker.dateOfBirth).format('MMM D, YYYY')}</MatchingField>
            <MatchingField type={getMatchType('email')}>
              <Typography>
                <a className={classes.emailAccent} href={`mailto:${existingWorker.email}`}>
                  {existingWorker.email}
                </a>
              </Typography>
            </MatchingField>
            {existingWorker.mobilePhoneNumber && currentWorker.mobilePhoneNumber && (
              <MatchingField type={getMatchType('mobilePhoneNumber')}>Mobile Phone: {formatPhoneNumber(existingWorker.mobilePhoneNumber)}</MatchingField>
            )}
            {existingWorker.socialSecurityNumber && currentWorker.socialSecurityNumber && (
              <MatchingField type={getMatchType('socialSecurityNumber')}>Last 4 SSN: {existingWorker.socialSecurityNumber}</MatchingField>
            )}
            {existingWorker.stateCode && currentWorker.address && (
              <MatchingField type={getMatchType('stateCode')}>State: {existingWorker.stateName}</MatchingField>
            )}
          </ul>
        </span>
      </div>
    </div>
  );
};

export default memo(DuplicatedWorkerModalContent);
