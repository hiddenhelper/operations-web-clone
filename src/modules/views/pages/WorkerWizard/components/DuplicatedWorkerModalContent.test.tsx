import React from 'react';
import { render } from '@testing-library/react';
import DuplicatedWorkerModalContent from './DuplicatedWorkerModalContent';
import { UserModel } from '../../../../models';
import { getExistingWorkerResponse, getWorker_1 } from '../../../../../test/entities';

describe('DuplicatedWorkerModalContent', () => {
  let props;

  beforeEach(() => {
    props = {
      matchedFields: ['fullName', 'email'],
      existingWorker: getExistingWorkerResponse().worker,
      currentWorker: { ...getWorker_1(), mobilePhoneNumber: '123456789' },
      userRole: UserModel.Role.FCA_ADMIN,
    };
  });

  it('should render', () => {
    const wrapper = render(<DuplicatedWorkerModalContent {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
