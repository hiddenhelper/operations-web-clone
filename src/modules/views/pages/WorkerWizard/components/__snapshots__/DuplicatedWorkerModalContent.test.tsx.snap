// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DuplicatedWorkerModalContent should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-duplicatedWorkerModalContent-40"
      >
        <p
          class="MuiTypography-root makeStyles-workerModalSecondaryTextPosition-41 makeStyles-secondaryText-9 MuiTypography-body1"
        >
          This information matches with an existing worker.
        </p>
        <div
          class="makeStyles-avatarWrapper-47"
        >
          <div>
            <div
              class="MuiAvatar-root MuiAvatar-circle makeStyles-workerModalAvatarPosition-42 makeStyles-avatarWrapper-11 makeStyles-workerMedium-26"
            >
              <img
                alt="Worker profile"
                class="MuiAvatar-img"
                src="https://cdn.business2community.com/wp-content/uploads/2017/08/blank-profile-picture-973460_640.png"
              />
            </div>
          </div>
          <span>
            <ul
              class="makeStyles-listWrapper-46"
            >
              <li
                class="match makeStyles-labelWrapper-52"
              >
                <div
                  class="makeStyles-workerModalIconMargin-51"
                >
                  <svg
                    class="makeStyles-workerModalIcon-50"
                  >
                    accepted.svg
                  </svg>
                </div>
                <p
                  class="MuiTypography-root makeStyles-workerAccent-44 MuiTypography-body1"
                />
              </li>
              <li
                class="noMatch makeStyles-labelWrapper-52"
              >
                <div
                  class="makeStyles-workerModalIconMargin-51"
                >
                  <svg
                    class="makeStyles-workerModalIcon-50"
                  >
                    expired.svg
                  </svg>
                </div>
                <p
                  class="MuiTypography-root makeStyles-companyAccent-45 MuiTypography-body1"
                >
                  Alpha Construction
                </p>
              </li>
              <li
                class="noMatch makeStyles-labelWrapper-52"
              >
                <div
                  class="makeStyles-workerModalIconMargin-51"
                >
                  <svg
                    class="makeStyles-workerModalIcon-50"
                  >
                    expired.svg
                  </svg>
                </div>
                Date of Birth: 
                Jan 1, 2001
              </li>
              <li
                class="match makeStyles-labelWrapper-52"
              >
                <div
                  class="makeStyles-workerModalIconMargin-51"
                >
                  <svg
                    class="makeStyles-workerModalIcon-50"
                  >
                    accepted.svg
                  </svg>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  <a
                    class="makeStyles-emailAccent-48"
                    href="mailto:test@test.com"
                  >
                    test@test.com
                  </a>
                </p>
              </li>
              <li
                class="noMatch makeStyles-labelWrapper-52"
              >
                <div
                  class="makeStyles-workerModalIconMargin-51"
                >
                  <svg
                    class="makeStyles-workerModalIcon-50"
                  >
                    expired.svg
                  </svg>
                </div>
                Mobile Phone: 
                1619123456
              </li>
              <li
                class="noMatch makeStyles-labelWrapper-52"
              >
                <div
                  class="makeStyles-workerModalIconMargin-51"
                >
                  <svg
                    class="makeStyles-workerModalIcon-50"
                  >
                    expired.svg
                  </svg>
                </div>
                Last 4 SSN: 
                123456789
              </li>
              <li
                class="noMatch makeStyles-labelWrapper-52"
              >
                <div
                  class="makeStyles-workerModalIconMargin-51"
                >
                  <svg
                    class="makeStyles-workerModalIcon-50"
                  >
                    expired.svg
                  </svg>
                </div>
                State: 
              </li>
            </ul>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-duplicatedWorkerModalContent-40"
    >
      <p
        class="MuiTypography-root makeStyles-workerModalSecondaryTextPosition-41 makeStyles-secondaryText-9 MuiTypography-body1"
      >
        This information matches with an existing worker.
      </p>
      <div
        class="makeStyles-avatarWrapper-47"
      >
        <div>
          <div
            class="MuiAvatar-root MuiAvatar-circle makeStyles-workerModalAvatarPosition-42 makeStyles-avatarWrapper-11 makeStyles-workerMedium-26"
          >
            <img
              alt="Worker profile"
              class="MuiAvatar-img"
              src="https://cdn.business2community.com/wp-content/uploads/2017/08/blank-profile-picture-973460_640.png"
            />
          </div>
        </div>
        <span>
          <ul
            class="makeStyles-listWrapper-46"
          >
            <li
              class="match makeStyles-labelWrapper-52"
            >
              <div
                class="makeStyles-workerModalIconMargin-51"
              >
                <svg
                  class="makeStyles-workerModalIcon-50"
                >
                  accepted.svg
                </svg>
              </div>
              <p
                class="MuiTypography-root makeStyles-workerAccent-44 MuiTypography-body1"
              />
            </li>
            <li
              class="noMatch makeStyles-labelWrapper-52"
            >
              <div
                class="makeStyles-workerModalIconMargin-51"
              >
                <svg
                  class="makeStyles-workerModalIcon-50"
                >
                  expired.svg
                </svg>
              </div>
              <p
                class="MuiTypography-root makeStyles-companyAccent-45 MuiTypography-body1"
              >
                Alpha Construction
              </p>
            </li>
            <li
              class="noMatch makeStyles-labelWrapper-52"
            >
              <div
                class="makeStyles-workerModalIconMargin-51"
              >
                <svg
                  class="makeStyles-workerModalIcon-50"
                >
                  expired.svg
                </svg>
              </div>
              Date of Birth: 
              Jan 1, 2001
            </li>
            <li
              class="match makeStyles-labelWrapper-52"
            >
              <div
                class="makeStyles-workerModalIconMargin-51"
              >
                <svg
                  class="makeStyles-workerModalIcon-50"
                >
                  accepted.svg
                </svg>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                <a
                  class="makeStyles-emailAccent-48"
                  href="mailto:test@test.com"
                >
                  test@test.com
                </a>
              </p>
            </li>
            <li
              class="noMatch makeStyles-labelWrapper-52"
            >
              <div
                class="makeStyles-workerModalIconMargin-51"
              >
                <svg
                  class="makeStyles-workerModalIcon-50"
                >
                  expired.svg
                </svg>
              </div>
              Mobile Phone: 
              1619123456
            </li>
            <li
              class="noMatch makeStyles-labelWrapper-52"
            >
              <div
                class="makeStyles-workerModalIconMargin-51"
              >
                <svg
                  class="makeStyles-workerModalIcon-50"
                >
                  expired.svg
                </svg>
              </div>
              Last 4 SSN: 
              123456789
            </li>
            <li
              class="noMatch makeStyles-labelWrapper-52"
            >
              <div
                class="makeStyles-workerModalIconMargin-51"
              >
                <svg
                  class="makeStyles-workerModalIcon-50"
                >
                  expired.svg
                </svg>
              </div>
              State: 
            </li>
          </ul>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
